cli/commands/upload/index.ts

```diff
@@ -0,0 +1,40 @@
+import { CommandHandler } from '../..'
+import { assertExists } from '../../utils'
+import { AppendToFile } from '../../utils/append.to.file'
+import { UploadManifest } from '../publish/upload.manifest'
+import { GetCredentials } from '../../utils/get.credentials'
+
+export default function provideUpload(
+  getCredentials: GetCredentials,
+  uploadManifest: UploadManifest,
+  appendToFile: AppendToFile,
+  args: any
+): CommandHandler {
+  assertExists(uploadManifest, 'uploadManifest')
+  assertExists(appendToFile, 'appendToFile')
+  assertExists(args, 'args')
+
+  return async function upload() {
+    if (!args.imageRef) {
+      console.error('--imageRef not provided - aborting')
+      return
+    }
+
+    let privateKey;
+    if (args.privateKey) {
```

Comment:
this is typically considered unsafe practice (i.e. requesting sensitive password/key in the command prompt because it gets logged to the CLI history and can be abused). i think we should remove this `privateKey` option. the current safe way to provide the key automatically is to specify it in the forta.config.json using `keyfile` and `keyfilePassword` (https://docs.forta.network/en/latest/config/)

Reply:
another alternative could be to support a `privateKey` field in the forta.config.json file

Reply:
> the current safe way to provide the key automatically is to specify it in the forta.config.json using `keyfile`

If the privatekey is a Github Actions Secret, would you recommend writing it to the keyfile from the environment variable in a workflow step?

Reply:
yea that could work too, as long as its written in the supported keyfile format (we use [keythereum](https://github.com/ethereumjs/keythereum) to generate keyfiles)

Reply:
Thanks @haseebrabbani, `keyfile` and `keyfilePassword` are very useful. Removed the `privateKey` arg.